device/components/RTA_OnDeviceComponentTask.brs

```diff
@@ -98,62 +89,63 @@ sub handleSocketEvent(message as Object)
 	' If the socketId matches our listen socketId this is a new connection being established
 	if messageSocketId = m.listenSocketId then
 		if m.listenSocket.isReadable() then
-			clientSocket = m.listenSocket.accept()
-			if clientSocket = Invalid then
+			socket = m.listenSocket.accept()
+			if socket = Invalid then
 				RTA_logError("Connection accept failed")
 			else
 				' We setup notification for when the new connection is readable
-				clientSocket.notifyReadable(true)
-				m.clientSockets[clientSocket.getID().toStr()] = clientSocket
+				socket.notifyReadable(true)
+				m.clientSockets[socket.getId().toStr()] = socket
 			end if
 		end if
 	else
-		clientSocket = m.clientSockets[messageSocketId]
-		bufferLength = clientSocket.getCountRcvBuf()
-		if bufferLength > 0 then
-			handleClientSocketEvent(messageSocketId, clientSocket, bufferLength)
-		else
-			RTA_logInfo("Client closed connection")
-			clientSocket.close()
-			m.clientSockets.delete(messageSocketId)
-		end if
+		' We are switching from recursive code to letting us receive another roSocketEvent where we can handle the different parts of the request
```

Comment:
This comment feels like it belongs in a github pull request comment rather than being in the code itself. The current code knows nothing about being recursive, so it might be worth just documenting what it actually does instead of what it used to do?